python/mxnet/runtime.py

```diff
@@ -69,12 +69,18 @@ def feature_list():
     features = [lib_features_c_array[i] for i in range(lib_features_size.value)]
     return features
 
+
 class Features(collections.OrderedDict):
     """
     OrderedDict of name to Feature
     """
+    feature_dict = None
```

Comment:
Since you are caching in ndarray.py now, I'd suggest we take this change out from PR. We can have a seperate PR to implement singleton properly as @larroy suggested.

Reply:
What is the concern here ? I don't see the harm of keeing it here. I will change it to singleton but caching results in individual files seems a makeshift approach. I would still prefer to keep it here for now till singleton is implemented

Reply:
@apeforest ^

Reply:
It's not needed in this PR and it's not related to what you are changing in this PR. It's always better to seperate features whenever unnecessary.